### PR TITLE
MapView shapshot and show all markers

### PR DESCRIPTION
**Pull request by ichan-mb**
Added two scriptclass functions, `snapshot` for taking an image of the current MapView display and `showAllMarkers` to zoom MapView to the particular position where all of the markers is visible on the MapView

**Examples:**

```
<App>
	<ClientPanel>
		<JavaScript>
			var Observable = require('FuseJS/Observable')
			var snapshotImage = new Observable()
			var Map = map;
			module.exports = {
				snapshotImage,
				showAllMarkers: function(args) {
					Map.showAllMarkers();
				},
				takeSnapshot: function(args) {
					Map.snapshot().then(function (path) {
						snapshotImage.value = path;
					})
				}
			};
		</JavaScript>
		<DockLayout />
		<NativeViewHost Dock="Top" Height="250">
			<MapView ux:Name="map" Zoom="14" Latitude="-6.914742" Longitude="107.609820">
				<MapMarker Latitude="-6.914742" Longitude="107.609820" />
				<MapMarker Latitude="-6.915850" Longitude="107.609929" />
				<MapMarker Latitude="-6.916959" Longitude="107.619009" />
			</MapView>
		</NativeViewHost>
		<Panel>
			<StackPanel ux:Name="stack" ItemSpacing="10" Orientation="Vertical">
				<Button Text="Show Markers" Clicked="{showAllMarkers}" />
				<Button Text="Snapshot" Clicked="{takeSnapshot}" />
			</StackPanel>
		</Panel>
		<Panel Dock="Bottom">
			<StackPanel ux:Name="a" ItemSpacing="10" Orientation="Vertical">
				<Image Size="200" File="{snapshotImage}" StretchMode="Uniform" />
				<Text Value="{snapshotImage}" />
			</StackPanel>
		</Panel>
	</ClientPanel>
</App>
```
This PR contains:
- [x] Changelog
- [x] Documentation
- [ ] Tests

### CODE REVIEW

File: Source/Fuse.Maps/MapView.ScriptClass.uno [hunk 1]
```diff
@@ -15,7 +15,9 @@ namespace Fuse.Controls
 				new ScriptMethod<MapView>("setBearing", setBearing),
 				new ScriptMethod<MapView>("setTilt", setTilt),
 				new ScriptMethod<MapView>("setZoom", setZoom),
+				new ScriptMethod<MapView>("showAllMarkers", showAllMarkers),
 				new ScriptMethod<MapView>("setMarkers", setMarkers),
+				new ScriptPromise<MapView,string,string>("snapshot", ExecutionThread.MainThread, snapshot, null),
```
Comment (mortend) [inline Source/Fuse.Maps/MapView.ScriptClass.uno:20]: Nit: Add space after comma in type parameter list, like so: `<MapView, string, string>`.

File: Source/Fuse.Maps/MapView.uno [hunk 2]
```diff
@@ -245,6 +247,18 @@ namespace Fuse.Controls
 				MapViewClient.UpdateOverlays();
 		}
 
+		void ShowAllMarkers()
+		{
+			if(MapIsReady)
```
Comment (mortend) [inline Source/Fuse.Maps/MapView.uno:252]: Nit: Add space after `if`, like so: `if (MapIsReady)`.

File: Source/Fuse.Maps/MapView.uno [hunk 3]
```diff
@@ -245,6 +247,18 @@ namespace Fuse.Controls
...
+				MapViewClient.ShowAllMarkers();
+		}
+
+		void Snapshot(Action<string> actionSucces, Action<string> actionError)
+		{
+			if(MapIsReady)
```
Comment (mortend) [inline Source/Fuse.Maps/MapView.uno:258]: Nit: Add space after `if`, like so: `if (MapIsReady)`.

File: Source/Fuse.Maps/iOS/MapViewDelegate.m [hunk 4]
```diff
@@ -528,12 +528,79 @@ -(MKCoordinateSpan)coordinateSpanWithMapView:(MKMapView *)mapView centerCoordina
...
+			}
+
+			UIImage *finalImage = UIGraphicsGetImageFromCurrentImageContext();
+			UIGraphicsEndImageContext();
+			NSData *data = UIImagePNGRepresentation(finalImage);
+			NSString * path = NSSearchPathForDirectoriesInDomains(NSLibraryDirectory, NSUserDomainMask, YES)[0];
+			NSString * filename = [path stringByAppendingPathComponent:@"map_snapshot.png"];
+			[data writeToFile:filename atomically:YES];
+
+			if(onSnapshotSucceed != nil)
```
Comment (mortend) [inline Source/Fuse.Maps/iOS/MapViewDelegate.m:601]: Nit: Add space after `if`, like so: `if (onSnapshotSucceed != nil)`.

**Review by mortend — COMMENTED** [review]
I want to nitpick a little bit about code style, and please rebase your branch to remove unnecessary merge commits. Thanks. :)

**Comment by ichan-mb** [comment]
closing and will create new PR with tidy up some commit

**Pull request closed without merging**